libs/ledger/include/ledger/chain/consensus/proof_of_work.hpp

```diff
@@ -70,24 +71,43 @@ inline vectorise::UInt<256> const &ProofOfWork::target() const
   return target_;
 }
 
-template <typename T>
-inline void Serialize(T &serializer, ProofOfWork const &p)
-{
-  serializer << p.header() << p.target();
-}
-
-template <typename T>
-inline void Deserialize(T &serializer, ProofOfWork &p)
-{
-  byte_array::ConstByteArray header;
-  vectorise::UInt<256>       target;
+}  // namespace consensus
+}  // namespace ledger
 
-  serializer >> header >> target;
+namespace serializers {
 
-  p.SetHeader(header);
-  p.SetTarget(std::move(target));
-}
+template <typename D>
+struct MapSerializer<ledger::consensus::ProofOfWork,
```

Comment:
I think this is fine for the moment. This data type will have a natural lifetime anyway